### PR TITLE
Implementation of Atomic tagging

### PR DESCRIPTION
**Pull request by sumitag**
Potential implementation using a flag for #360

### CODE REVIEW

**Comment by coderanger** [comment]
There's not really any reason to introduce a new option. It should use in-stream tags for non-spot instance and dumb-slow-tags for spot instances.

(also almost everyone should probably be using spot instances for kitchen-ec2 which is part of why I've not really bothered fixing this)

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ cache: bundler
 sudo: false
 branches:
   only:
-  - master
+  - atomic-tags
```
Comment (coderanger) [inline .travis.yml]: This doesn't work, also never do this in a FOSS project.

File: kitchen-ec2.gemspec [hunk 2]
```diff
@@ -23,7 +23,7 @@ Gem::Specification.new do |gem|
   gem.add_dependency "test-kitchen", "~> 1.4", ">= 1.4.1"
   gem.add_dependency "excon"
   gem.add_dependency "multi_json"
-  gem.add_dependency "aws-sdk", "~> 2"
+  gem.add_dependency "aws-sdk", "~> 3"
```
Comment (coderanger) [inline kitchen-ec2.gemspec]: This needs to be x-ref'd with ChefDK.

File: .travis.yml (hunk 1 above)
Comment (sumitag) [reply .travis.yml]: Absolutely, I was just running through travis for my local branch and forgot to revert it.

File: kitchen-ec2.gemspec (hunk 2 above)
Comment (sumitag) [reply kitchen-ec2.gemspec]: ChefDK is using ~> 2, I am going to change it back.

**Comment by sumitag** [comment]
At our work we are using normal instances and not spot instances and also want to restrict instance creations with valid tags using IAM.

I will implement in-stream tags for non spot instances.

**Comment by sumitag** [comment]
@coderanger I have updated the branch, can you please have a look?

**Comment by clintoncwolfe** [comment]
I'm looking to continue this work; @sumitag do you mind if I adopt this PR?

**Comment by sumitag** [comment]
Please feel free to use the PR

**Comment by tas50** [comment]
This has been resolved with https://github.com/test-kitchen/kitchen-ec2/pull/496

**Pull request closed without merging**